storey/sources.py

```diff
@@ -875,3 +875,124 @@ def _init(self):
             else:
                 df = pandas.read_parquet(path, columns=self._columns, storage_options=self._storage_options)
             self._dfs.append(df)
+
+    class MongoDBSource(_IterableSource, WithUUID):
+        """Use mongodb collection as input source for a flow.
+
+        :parameter key_field: string. column to be used as key for events. can be list of columns
+        :parameter time_field: string. column to be used as time for events.
+        :parameter id_field: string. column to be used as ID for events.
+        :parameter db_name: string. the name of the database you would like to access as it mention on mongodb
+        :parameter connection_string: string. your mongodb connection string
+        :parameter collection_name: string. the collection name you would like to access,
+                                    from the current db as it mention on mongodb
+        :parameter query: dict. dictionary query for mongodb
+        :parameter start_filter: datetime. If not None, the results will be filtered by partitions and 'filter_column' > start_filter.
+                                Default is None
+        :parameter end_filter: datetime. If not None, the results will be filtered by partitions 'filter_column' <= end_filter.
+                                Default is None
+
+        """
+
+        def __init__(
+                self,
+                db_name: str = None,
+                connection_string: str = None,
+                collection_name: str = None,
+                query: dict = None,
+                key_field: Optional[Union[str, List[str]]] = None,
+                start_filter: Optional[datetime] = None,
+                end_filter: Optional[datetime] = None,
+                time_field: Optional[str] = None,
+                id_field: Optional[str] = None,
+                **kwargs,
+        ):
+
+            if time_field:
+                time_query = {time_field: None}
+                if start_filter:
+                    time_query[time_field] = {"$gte": start_filter}
+                if end_filter:
+                    time_query[time_field]["$lt"] = end_filter
```

Comment:
I'm pretty sure this will raise an error if you pass an `end_filter` but no `start_filter`.